db/migrate/20241205142318_add_more_native_columns_to_contacts.rb

```diff
@@ -0,0 +1,21 @@
+class AddMoreNativeColumnsToContacts < ActiveRecord::Migration[8.0]
+  def change
+    change_table :contacts do |t|
+      t.string :status, null: false, default: 'active'
+      t.string :language_code
+      t.string :gender
+      t.date :date_of_birth
+
+      t.string :iso_country_code
+      t.string :iso_region_code
+      t.string :administrative_division_level_2_code
+      t.string :administrative_division_level_2_name
+      t.string :administrative_division_level_3_code
+      t.string :administrative_division_level_3_name
+      t.string :administrative_division_level_4_code
+      t.string :administrative_division_level_4_name
+
+      t.index :metadata, using: :gin
+    end
```

Comment:
You should add a combined index here for all the location columns e.g. [iso_country_code, iso_region_code3, …]

Reply:
Those columns are optional, so if we add indexes, we need to consider multiple combinations since the code can be blank. Maybe we just add only for the code not the name.

index: iso_country_code, iso_region_code, administrative_division_level_2_code

Reply:
Something like this probably:

```rb
add_index :contacts, [:iso_country_code, :iso_region_code, :administrative_division_level_2_code, administrative_division_level_3_code, :administrative_division_level_4_code]

add_index :contacts, [:iso_country_code, :iso_region_code, administrative_division_level_2_name, administrative_division_level_3_name, administrative_division_level_4_name]
```